sdk/keyvault/Azure.Security.KeyVault.Keys/src/Key.cs

```diff
@@ -64,7 +64,7 @@ internal virtual void ReadProperty(JsonProperty prop)
                 case KeyPropertyName:
                     KeyMaterial = new JsonWebKey();
                     KeyMaterial.ReadProperties(prop.Value);
-                    Properties.ParseId(KeyMaterial.KeyId);
+                    Properties.ParseId(KeyMaterial.Id);
```

Comment:
>Properties.ParseId(KeyMaterial.Id); [](start = 20, length = 35)

I believe kid is also returned directly on the key. If so this might overwrite the value which is passed back as the top level kid.

Reply:
At least on https://docs.microsoft.com/en-us/rest/api/keyvault/createkey/createkey#jsonwebkey and https://docs.microsoft.com/en-us/rest/api/keyvault/getkey/getkey#jsonwebkey, the `kid` is only on the JWK so this would be necessary. If we look at https://docs.microsoft.com/en-us/rest/api/keyvault/getkeys/getkeys#keylistresult `kid` is a top-level property but is returned from `KeyClient` as a list of `KeyProperties`.